### ISSUE TITLE
SystemColorConstants.vbHotTrackText

### ISSUE THREAD

**Issue by Kr00l** [WaynePhillipsEA/twinbasic#571]
**Is your feature request related to a problem? Please describe.**
I use the sys color COLOR_HOTLIGHT quite often to make text "blue" when mouse hovers over an element.
It would be convenient to have this just included into the SystemColorConstants.

**Describe the solution you'd like**
Add enum.
```
Const COLOR_HOTLIGHT As Long = 26
Const vbHotTrackText = &H8000001A
```

**Describe alternatives you've considered**
existing vbHighlightText is not applicable as this is "white" (COLOR_HIGHLIGHTTEXT) and not applicable for hot-tracking.

**Additional context**
The naming 'HotTrackText' is copied from 
[colors-and-styling-for-visual-studio](https://docs.microsoft.com/en-us/visualstudio/extensibility/ux-guidelines/colors-and-styling-for-visual-studio?view=vs-2022) where it states..
_WindowText, GrayText and HotTrackText are the only colors available on WindowBackground surfaces._

And in fact GetSysColor API describes it also 
_Color for a hyperlink or hot-tracked item. The associated background color is COLOR_WINDOW._

That's why I think vbHotTrackText is the best name for this enum.

**Comment by wqweto** [comment]
Btw, these both fail
```
    Call OleTranslateColor(vbHotTrackText, 0, clrFore)
    Label2.ForeColor = vbHotTrackText
```
. . . but this works
```
    clrFore = GetSysColor(vbHotTrackText And &HFF)
```

**Comment by Kr00l** [comment]
ok. what a shame. So I may close this issue ?

**Comment by wqweto** [comment]
I wish it work because it's a cool color to have.

**Comment by Kr00l** [comment]
OleTranslateColor is obselete. So the only thing I see is that tB don't use this (also for cross-platform it would be wise) and has it's own function.
So, that at least vbHotTrackText can be assigned to OLE_COLOR properties. Of course exposing this replacing OleTranslateColor in the VBA library would be nice.

**Comment by wqweto** [comment]
Btw, I didn't know that [dilletante can mark Win32 API functions obsolete directly with posts in vbforums.com](https://www.vbforums.com/showthread.php?871651-ARGB-color-values&p=5362667&viewfull=1#post5362667) but always though this status must be documented on MSDN somewhere :-))

At least when talking about the technical term "obsolete" that is.

**Comment by Kr00l** [comment]
Looking at [reactOS implementation](https://doxygen.reactos.org/df/d5c/oleaut_8c.html#a0c970cce1194cdc1f3a188ff7bc1f63f)  there is a syscolor validation between COLOR_SCROLLBAR (0) and COLOR_MENUBAR (30)
So, when tB would implement below function it could handle COLOR_HOTLIGHT. :) So they do better at this than MS.

```
 /******************************************************************************
  *      OleTranslateColor   [OLEAUT32.421]
  *
  * Convert an OLE_COLOR to a COLORREF.
  *
  * PARAMS
  *  clr       [I] Color to convert
  *  hpal      [I] Handle to a palette for the conversion
  *  pColorRef [O] Destination for converted color, or NULL to test if the conversion is ok
  *
  * RETURNS
  *  Success: S_OK. The conversion is ok, and pColorRef contains the converted color if non-NULL.
  *  Failure: E_INVALIDARG, if any argument is invalid.
  *
  * FIXME
  *  Document the conversion rules.
  */
 HRESULT WINAPI OleTranslateColor(
   OLE_COLOR clr,
   HPALETTE  hpal,
   COLORREF* pColorRef)
 {
   COLORREF colorref;
   BYTE b = HIBYTE(HIWORD(clr));
 
   TRACE("(%08x, %p, %p)\n", clr, hpal, pColorRef);
 
   /*
    * In case pColorRef is NULL, provide our own to simplify the code.
    */
   if (pColorRef == NULL)
     pColorRef = &colorref;
 
   switch (b)
   {
     case 0x00:
     {
       if (hpal != 0)
         *pColorRef =  PALETTERGB(GetRValue(clr),
                                  GetGValue(clr),
                                  GetBValue(clr));
       else
         *pColorRef = clr;
 
       break;
     }
 
     case 0x01:
     {
       if (hpal != 0)
       {
         PALETTEENTRY pe;
         /*
          * Validate the palette index.
          */
         if (GetPaletteEntries(hpal, LOWORD(clr), 1, &pe) == 0)
           return E_INVALIDARG;
       }
 
       *pColorRef = clr;
 
       break;
     }
 
     case 0x02:
       *pColorRef = clr;
       break;
 
     case 0x80:
     {
       int index = LOBYTE(LOWORD(clr));
 
       /*
        * Validate GetSysColor index.
        */
       if ((index < COLOR_SCROLLBAR) || (index > COLOR_MENUBAR))
         return E_INVALIDARG;
 
       *pColorRef =  GetSysColor(index);
 
       break;
     }
 
     default:
       return E_INVALIDARG;
   }
 
   return S_OK;
 }
```

**Comment by WaynePhillipsEA** [comment]
VBA.Information.TranslateColor is now available in the standard library (v0.14.1).  vbHotTrackText will follow in the next update

**Comment by WaynePhillipsEA** [comment]
This is now available in v0.14.2.  Thanks!

**Comment by Kr00l** [comment]
Thanks.
However, at the risk of being labelled pedantic. :D

COLOR_ACTIVECAPTION is the left side starting color and COLOR_GRADIENTACTIVECAPTION is the right side ending color.
In VBx the starting color is vbActiveTitleBar
IMO tB shall name the enum `vbGradientActiveTitleBar` or `vbActiveTitleBarGradient` instead of `vbGradientActiveCaption`.
Same with `vbInactiveTitleBar`. Shall be `vbGradientInactiveTitleBar` or `vbInactiveTitleBarGradient` instead of `vbGradientInactiveCaption`.
So they relate together. :)

**Comment by Kr00l** [comment]
Following naming logic vbInfoText and vbInfoBackground (to tie them together) it is probably best to name `vbActiveTitleBarGradient` and `vbInactiveTitleBarGradient` to tie them together under each.

**Comment by Kr00l** [comment]
I re-opened but close now again as Wayne should decide anyway, so it's in normal process now.

**Comment by WaynePhillipsEA** [comment]
That's fine @Kr00l feel free to re-open things if they need my attention.  

Fixed as suggested in v0.14.3.  Thanks!
